### ISSUE TITLE
cpan Firefox::Marionette failures

### ISSUE THREAD

**Issue by Veraellyunjie** [david-dick/firefox-marionette#68]
### Case 1 - Windows 10

`cpan Firefox::Marionette` fails [cpan_Firefox_Marionette_Windows10.txt](https://github.com/user-attachments/files/17563197/cpan_Firefox_Marionette_Windows10.txt)

installed with `cpan -T Firefox::Marionette` skipping tests

copy-paste [SYNOPSIS](https://metacpan.org/pod/Firefox::Marionette#SYNOPSIS) into `mario.pl` [mario.txt](https://github.com/user-attachments/files/17563218/mario.txt) (renamed as `mario.txt` for github to accept it for upload)

run `perl mario.pl` - seems to work ok [mario_Windows10.txt](https://github.com/user-attachments/files/17563229/mario_Windows10.txt)

---

### Сase 2 - OpenBSD 7.6

`cpan Firefox::Marionette` fails [cpan_Firefox_Marionette_OpenBSD76.txt](https://github.com/user-attachments/files/17563249/cpan_Firefox_Marionette_OpenBSD76.txt)

installed with `cpan -T Firefox::Marionette` skipping tests

copy-paste [SYNOPSIS](https://metacpan.org/pod/Firefox::Marionette#SYNOPSIS) into `mario.pl` [mario.txt](https://github.com/user-attachments/files/17563218/mario.txt) (renamed as `mario.txt` for github to accept it for upload)

run `perl mario.pl` with proxy env vars
```
> env | grep -i proxy | grep -vi no_proxy | sort
ALL_PROXY=socks://127.0.0.1:9050
FTPS_PROXY=socks://127.0.0.1:9050
FTP_PROXY=socks://127.0.0.1:9050
HTTPS_PROXY=socks://127.0.0.1:9050
HTTP_PROXY=socks://127.0.0.1:9050
all_proxy=socks://127.0.0.1:9050
ftp_proxy=socks://127.0.0.1:9050
ftps_proxy=socks://127.0.0.1:9050
http_proxy=socks://127.0.0.1:9050
https_proxy=socks://127.0.0.1:9050
```
\- it fails
`Can't locate object method "host_port" via package "URI::_foreign" at /usr/local/libdata/perl5/site_perl/Firefox/Marionette.pm line 8517.`

run `perl mario.pl` without proxy - it fails
`Firefox killed by a SEGV signal (11) at mario.pl line 4`

---

So,
fails tests but seems to work on Windows,
fails tests and doesn't work on OpenBSD.
`cpan` says there are missing modules, but doesn't it take care about such thing installing everything needed?

---

Do I install and run it right?
Are there any installation notes, e.g. root-vs-user on unix-like or Windows peculiarities? I haven't found them

**Comment by david-dick** [comment]
Looks like your version of firefox on windows has something weird stopping an element from being fully shown in the screen.

With the socks proxy errors, try to install LWP::protocol::socks first.  That should help.

**Comment by david-dick** [comment]
Looks like the firefox package in OpenBSD 7.6 SEGVs when it is not run under Xvfb.  Try running with the visible flag set on.

**Comment by david-dick** [comment]
`my $firefox = Firefox::Marionette->new( visible => 1)->go('https://metacpan.org/');`

**Comment by david-dick** [comment]
The issues raised here should be resolved with the 1.62 release, just pushed to CPAN.  Please reopen if this is not the case.
